open-craft/xblock-ai-evaluation
# feat: create xblocks

### Testing instructions

1. Clone the repository and switch to the PR branch `create-xblocks`.
2. Add it as a tutor mount `tutor mounts add /workspace/xblock-ai-completion`
3. Rebuild the Docker image `tutor images build openedx-dev`
4. Start tutor `tutor dev start`
5. Install the XBlock inside the LMS and CMS containers
    ```
    tutor dev run cms bash
    pip install -e /mnt/xblock-ai-evaluation

    tutor dev run lms bash
    pip install -e /mnt/xblock-ai-evaluation
    ```
6. Restart the LMS and CMS containers
    ```
    tutor dev restart cms
    tutor dev restart lms
    ```
7. In the Open edX platform, navigate to `Settings > Advanced Settings` and add `shortanswer_ai_eval` and `coding_ai_eval` to the `Advanced Module List`.
8. Add either XBlock using the `Advanced` button in the `Add New Component` section of Studio.
9. Configure the added Xblock and make sure to add correct API keys. You can format your question and prompts using [Markdown](https://marked.js.org/demo/).
